### PR TITLE
Add documentation that explains how to add project to CanJS site

### LINKED ISSUE #2844: Add docs for adding to the CanJS ecosystem
Please add instructions for how to get a package included in the CanJS website in the ecosystem sections.

Related to https://github.com/canjs/canjs/issues/2463

### PR DESCRIPTION
**Pull request by Macrofig**
Resolves #2844.

Added docs for including a project's docs in CanJS via ecosystem or legacy.

I'd appreciate feedback on structure, document title, document location, and any spelling/grammar mistakes!

Only concern I have is that bit-docs tries to process a tag that is in a code block.

```text
@ module {Object} my-project
@ parent can-ecosystem
@ package ../package.json

> Documentation goes here
```

image

A somewhat pretty version of the markdown can be found here: https://gist.github.com/Macrofig/32b4070858b4dd1bee97a40b86f212bc

### CODE REVIEW

File: docs/can-guides/contribute/including-new-modules.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+@page guides/contributing/include-new-modules Including New Modules
```
Comment (justinbmeyer) [inline docs/can-guides/contribute/including-new-modules.md]: This should probably be `Adding Ecosystem Packages` as that will be by far the most common.

File: docs/can-guides/contribute/including-new-modules.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+cd my-fancy-utility
+touch my-fancy-utility.md
+```
+
+## Source Repo Documentation Tags
+
+In order to include the source project's documentation on the CanJS site, the main document file requires some tags.
+
+```javadocs
+@ module {Object} <MODULE_NAME>
```
Comment (justinbmeyer) [inline docs/can-guides/contribute/including-new-modules.md]: You can do `@@` I think.

Comment (justinbmeyer) [inline docs/can-guides/contribute/including-new-modules.md]: Also, might not be an object.  But it's ok.

Probably should be `<PACKAGE_NAME>`

File: docs/can-guides/contribute/including-new-modules.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+@body
+
+A project that is useful to the CanJS community can be included on the [CanJS documentation](http://canjs.com/) site.
+
+Before submitting a PR with these changes, make sure that the source module has [guides/contributing/documentation] and tests that follow [guides/contributing/code] standards set forth by the CanJS community.
+
+## Source Repo Documentation
+
+Create a markdown file in the `docs` folder of source repo and name it after the project.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: “source repo” isn’t defined before this section, so I would either define it in the intro (e.g. ‘In this guide, we’ll use the term “source repo” to refer to the repository of the package that you want to add to the ecosystem collection’) or make it more clear here (e.g. …‘folder in your package’s repository and’…).

File: docs/can-guides/contribute/including-new-modules.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+```bash
+# This example puts the documents in the project root
+cd my-fancy-utility
+touch my-fancy-utility.md
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: I’m a little confused about this because the instructions above say you should create a `docs` folder but this puts it in the root.

File: docs/can-guides/contribute/including-new-modules.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+In order to include the source project's documentation on the CanJS site, the main document file requires some tags.
+
+```javadocs
+@ module {Object} <MODULE_NAME>
+@ parent <CANJS_PARENT_TARGET>
+@ package <PATH_TO_PACKAGE.JSON>
+```
+
+- `MODULE_NAME`: The project name, which should match the `name` property in the source project's `package.json`.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: You can make this consistent with the other list items by replacing `: The project` with `: is the project`

File: docs/can-guides/contribute/including-new-modules.md (hunk 1 above)
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: Maybe “Adding Packages to CanJS.com”? I’m weary of putting “ecosystem” in the name because it applies to all of the collections.

File: docs/can-guides/contribute/including-new-modules.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+- `CANJS_PARENT_TARGET`: is the name of the target area in CanJS. For example, if the module is a part of the "Ecosystem", then the value for `@parent` would be `can-ecosystem`. If the target is "Legacy" than the value is `can-legacy`.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: “than” => “then”

File: docs/can-guides/contribute/including-new-modules.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+Open the desired file and require the source project.
+
+```javascript
+require('<SOURCE_MODULE_NAME>');
+```
+
+## Require source tests in CanJS repo
+
+Open the test main file in the CanJS repo, located at `canjs/test/test.js`.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: “test main file” => “main test file”? Here and in the sentence below.

File: docs/can-guides/contribute/including-new-modules.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+Require the source repo's test main file in the appropriate area. For example, if source repo will go in "Legacy" area of the site, then add it to the "Legacy" section of the test file.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: “if source” => “if the source”

File: docs/can-guides/contribute/including-new-modules.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+- `MODULE_NAME`: The project name, which should match the `name` property in the source project's `package.json`.
+- `MODULE_DESCRIPTION`: short description of project, can match the `description` property in source project's `package.json`.
+- `PROJECT_GITHUB_URL`: The Github url. Not the `git` path but the path to the html site.
+- `PROJECT_GITHUB_ORG`: The organization id or user id that owns the project in Github.
+
+
+## Notes
+
+The last part of the process is to test the documentation and test process locally. Once the site build is verified, create a PR against the CanJS repo to submit changes.
```
Comment (chasenlehara) [inline docs/can-guides/contribute/including-new-modules.md]: This wording is a little hard to parse: “to test the documentation and test process locally”

Suggestion: “to build the documentation and run the tests locally”

Also, consider “in” instead of “against”.

File: docs/can-guides/contribute/including-new-modules.md (hunk 1 above)
Comment (Macrofig) [inline docs/can-guides/contribute/including-new-modules.md]: I tried not to mention "ecosystem" exclusively because it should be helpful for every list.  As a compromise, though, I tried to use examples specific to "ecosystem".

File: docs/can-guides/contribute/including-new-modules.md (hunk 2 above)
Comment (Macrofig) [inline docs/can-guides/contribute/including-new-modules.md]: D'oh. I knew there was a better name than "module name". Thanks!

File: docs/can-guides/contribute/including-new-modules.md (hunk 6 above)
Comment (Macrofig) [inline docs/can-guides/contribute/including-new-modules.md]: Ohhhh... I can't believe I missed that!

File: docs/can-guides/contribute/including-new-modules.md (hunk 4 above)
Comment (Macrofig) [inline docs/can-guides/contribute/including-new-modules.md]: Actually, it says...
> Create a markdown file in the `docs` folder of source repo

But I was afraid that wouldn't be clear so I'm glad you stumbled there, though. I'll fix it.

**Comment by justinbmeyer** [comment]
Ecosystem is 99% of what people will be adding plugins to going forward. The people who will be adding to the other parts are likely only going to be the core team. Generally speaking, we will just be moving something incubated in ecosystem to core.

**Comment by justinbmeyer** [comment]
One other thing to add is `TravisCI` in the repo.

**Comment by justinbmeyer** [comment]
This is a good start and should be merged.  It could use some additional info about how to integrate into the build, but that can be done later.

Also, the ecosystem landing page should link to this.

**Comment by chasenlehara** [comment]
Thanks @Macrofig! I’ve merged this and created another issue for us to follow up with: https://github.com/canjs/canjs/issues/3055
